CMakeLists.txt

```diff
@@ -42,11 +43,25 @@ option(USE_JSON "Enable use of JSON library" ON)
 option(USE_HTTPLIB "Enable use of httplib" ON)
 option(USE_GRPC "Enable GRPC and Protobuf" ON)
 option(USE_OPENSSL "Enable use of OpenSSL" ON)
+option(USE_EDDSA_OPENSSL "Enable use of EdDSA's OpenSSL implementation for signature generation and verification" ON)
 option(BUILD_THIRDPARTY "Wheter to build third party librarie or use preinstalled ones" OFF)
 option(CODECOVERAGE "Enable Code Coverage Metrics in Clang" OFF)
 option(ENABLE_RESTART_RECOVERY_TESTS "Enable tests for restart recovery" OFF)
 
-if(USE_OPENSSL AND NOT BUILD_THIRDPARTY)
+if((NOT (USE_EDDSA_OPENSSL OR USE_RELIC)) OR (USE_EDDSA_OPENSSL AND USE_RELIC))
```

Comment:
maybe put this logic down to the threshold CMakeLists.txt?
there's no mention of threshold in these options, currently we still can use both relic and openssl eddsa instead of rsa